src/lazy/expanded/macro_evaluator.rs

```diff
@@ -642,31 +750,226 @@ mod tests {
     ///
     /// This test exists to demonstrate that macro evaluation within the TDL context works the
     /// same as evaluation in the data stream.
-    fn eval_tdl_template_invocation(invocation: &str, expected: &str) -> IonResult<()> {
-        let macro_table = MacroTable::new();
-        let symbol_table = SymbolTable::new();
-        let allocator = BumpAllocator::new();
-        let context = EncodingContext::new(&macro_table, &symbol_table, &allocator);
-        let mut evaluator = TdlMacroEvaluator::<TextEncoding_1_1>::new();
-        let invocation = Element::read_one(invocation)?;
-        let actuals = evaluator.evaluate(context, invocation.expect_sexp()?)?;
+    fn eval_template_invocation(
+        template_definition: &str,
+        invocation: &str,
+        expected: &str,
+    ) -> IonResult<()> {
+        let mut reader = LazyTextReader_1_1::new(invocation.as_bytes())?;
+        let _macro_address = reader.register_template(template_definition)?;
+        let actuals = reader.read_all_elements()?;
```

Comment:
🗺️ We no longer have to manually construct an `EncodingContext` and build an evaluator. We can just use a `LazyTextReader_1_1`. For now, though, we have to manually hand it our template definition text.